cpp/dolfinx/la/SparsityPattern.cpp

```diff
@@ -140,14 +140,29 @@ SparsityPattern::SparsityPattern(
   }
 }
 //-----------------------------------------------------------------------------
+void SparsityPattern::insert(std::int32_t row, std::int32_t col)
```

Comment:
It it, but the interface of the caller becomes cleaner. Instead of needing to call 
```
sp.insert(std::vector<std::int32_t>(row), std::vector<std::int32_t>(col));
```
 we can now simply call
```
sp.insert(row, col);
```
same holds for the python interface with `numpy` arrays.

Also, in combination with the span based version it makes clearer how the sparsity pattern is to be used, i.e. loop over entries and dense matrices and do not pre-compute the entries and then pass it in bulk. Sparsity pattern already performs the caching and associated memory management.

Reply:
Well, it's clear from the improved docstring. I don't see the point of this new method, the clarity of overloading functions is debatable.

Reply:
To add, the interface is vectorised from the Python end to encourage users to pass in multiple entries, avoiding the overhead of the function call.

Reply:
How/where is the interface vectorized? I can not find that.

Reply:
@chrisrichardson mentioned the other day, we might instead of the integer version consider to create a non owning `const std::vector<std::int32_t>&` version, this should allow for the interface to be `insert({row}, {col})`, i.e. it allows for conversion from an initializer list.